### PR TITLE
include field paths in unknown/duplicate errors

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Part of **https://github.com/kubernetes/enhancements/issues/2885**

/milestone v1.24

```release-note
NONE
```

/sig api-machinery
/cc @kevindelgado @apelisse

### CODE REVIEW

File: vendor/sigs.k8s.io/json/internal/golang/encoding/json/kubernetes_patch.go [hunk 1]
```diff
@@ -69,6 +71,14 @@ func (d *Decoder) DisallowDuplicateFields() {
...
+	if len(d.strictFieldStack) > 0 {
+		return fmt.Errorf("%s %q", msg, strings.Join(d.strictFieldStack, "")+"."+field)
+	} else {
+		return fmt.Errorf("%s %q", msg, field)
+	}
+}
+
 // saveStrictError saves a strict decoding error,
 // for reporting at the end of the unmarshal if no other errors occurred.
 func (d *decodeState) saveStrictError(err error) {
```
Comment (kevindelgado) [inline vendor/sigs.k8s.io/json/internal/golang/encoding/json/kubernetes_patch.go:84]: Do you think we should add a similar excessive-number-of-errors to to the pruning and conversion error accumulations I added?

**Review by kevindelgado — COMMENTED** [review]
One small question this raised about the code I worked on

/lgtm

File: vendor/sigs.k8s.io/json/internal/golang/encoding/json/kubernetes_patch.go (hunk 1 above)
Comment (liggitt) [reply vendor/sigs.k8s.io/json/internal/golang/encoding/json/kubernetes_patch.go:84]: mm... we could... I don't feel as strongly about it there as in a low-level decoding library

**Comment by liggitt** [comment]
rebased

**Comment by liggitt** [comment]
rebased

**Pull request merged**